core/domain/collection_services.py

```diff
@@ -526,6 +531,14 @@ def _save_collection(committer_id, collection, commit_message, change_list):
     else:
         collection.validate(strict=False)
 
+    for collection_node in collection.nodes:
+        if not exp_services.get_exploration_by_id(
+            collection_node.exploration_id, strict=False):
+            raise Exception(
```

Comment:
Done. 
Also can you clarify something. Why is the changelist error above this not utils.ValidationError?